index.html

```diff
@@ -226,6 +231,10 @@ <h2>
         steps:
       </p>
       <ol>
+        <li>
```

Comment:
How about if we pass `this` in the `vibrate()` method steps to 'perform vibration' algorithm so that it has access? PR updated.

(I recall there was some good reason to keep 'perform vibration' separate, so I try to keep it that way by passing `this` around.)

Reply:
Weird yes! Valid, maybe..? Here's my logic (and please correct me):

Per https://webidl.spec.whatwg.org/#this `this` is exposed to constructor steps, getter steps, setter steps, and method steps, so algorithms like "perform vibration" won't have access to `this` unless it is passed as an input variable to the algorithm.

That said, even if this would be valid it is not clear, so I'll adopt your proposal to pass "this's relevant global object" instead. Clear always wins smart in spec-land :)